packages/protocol/test/governance/epochrewards.ts

```diff
@@ -0,0 +1,576 @@
+    })
+  })
+
+  describe('#setTargetVotingYieldParameters()', () => {
+    describe('when the parameters are different', () => {
+      const newMax = targetVotingYieldParams.max.plus(1)
+      const newFactor = targetVotingYieldParams.adjustmentFactor.plus(1)
+
+      describe('when called by the owner', () => {
+        let resp: any
+
+        beforeEach(async () => {
+          resp = await epochRewards.setTargetVotingYieldParameters(newMax, newFactor)
+        })
+
+        it('should set the new target voting yield parameters', async () => {
+          const [, max, adjustmentFactor] = await epochRewards.getTargetVotingYieldParameters()
+          assertEqualBN(max, newMax)
+          assertEqualBN(adjustmentFactor, newFactor)
+        })
+
+        it('should emit the TargetVotingYieldParametersSet event', async () => {
+          assert.equal(resp.logs.length, 1)
+          const log = resp.logs[0]
+          assertContainSubset(log, {
+            event: 'TargetVotingYieldParametersSet',
+            args: {
+              max: newMax,
+              adjustmentFactor: newFactor,
+            },
+          })
+        })
+
+        describe('when called by a non-owner', () => {
+          it('should revert', async () => {
+            await assertRevert(
+              epochRewards.setTargetVotingYieldParameters(newMax, newFactor, {
+                from: nonOwner,
+              })
+            )
+          })
+        })
+      })
+
+      describe('when the parameters are the same', () => {
+        it('should revert', async () => {
+          await assertRevert(
+            epochRewards.setTargetVotingYieldParameters(
+              targetVotingYieldParams.max,
+              targetVotingYieldParams.adjustmentFactor
+            )
+          )
+        })
+      })
+    })
+  })
+
+  describe('#getTargetGoldTotalSupply()', () => {
+    describe('when it has been fewer than 15 years since genesis', () => {
+      const timeDelta = YEAR.times(10)
+      beforeEach(async () => {
+        await timeTravel(timeDelta.toNumber(), web3)
+      })
+
+      it('should return 600MM + 200MM * t / 15', async () => {
+        assertEqualBN(
+          await epochRewards.getTargetGoldTotalSupply(),
+          getExpectedTargetTotalSupply(timeDelta)
+        )
+      })
+    })
+  })
+
+  describe('#getTargetEpochRewards()', () => {
+    describe('when there are active votes', () => {
+      const activeVotes = 1000000
+      beforeEach(async () => {
+        await mockElection.setActiveVotes(activeVotes)
+      })
+
+      it('should return a percentage of the active votes', async () => {
+        const expected = fromFixed(targetVotingYieldParams.initial).times(activeVotes)
+        assertEqualBN(await epochRewards.getTargetEpochRewards(), expected)
+      })
+    })
+  })
+
+  describe('#getTargetTotalEpochPaymentsInGold()', () => {
+    describe('when a StableToken exchange rate is set', () => {
+      // Hard coded in EpochRewardsTest.sol
```

Comment:
Would reasoning about this code/writing potential future tests be clearer if we allowed setting `numValidators` in the mock contract and did that here?